index.js

```diff
@@ -1,32 +1,45 @@
-const client = require('./lib/statsd');
+/*
+ * Initialize a hot-shots client.
+ */
 
-const isNotEmpty = (obj) => obj && Object.getOwnPropertyNames(obj).length;
+const os = require('os');
 
-function keyValuesToString(keyValues) {
-  return Object.keys(keyValues)
-    .reduce((r, k) => r.concat(`${k}=${keyValues[k]}`), [])
-    .join(',');
+const statsdClient = require('hot-shots');
+const strigoLogger = require('strigo-node-logger');
+
+const STATSD_HOST = '127.0.0.1';
+const STATSD_PORT = 8125;
+const DEFAULT_ENV = os.hostname();
+const CONTAINER_HOSTNAME = process.env.HOSTNAME || 'null';
```

Comment:
how is `process.env.HOSTNAME` differs from `os.hostname()`?

what does the container_hostname means? shouldn't it just be the hostname?

Reply:
That's exactly right. If it's not in docker, it should be null.

Reply:
And Btw, the hostname inside a container isn't random, but rather is a substring of the ID.

Reply:
https://docs.docker.com/config/containers/container-networking/#ip-address-and-hostname

Reply:
but id is random most of the time so it's sort of random. point is, i think we should have something to correlate

Reply:
What? No.. in Docker the hostname is the container ID. I think it makes sense to have it, but I agree we should have the alloc ID. You think I should read that from the env var nomad exposes to the process?

Reply:
Ok, wait I got it wrong. For some reason I thought that `HOSTNAME` envar was not set in a docker container... 

`NOMAD_ALLOC_ID` is the nomad allocation.

A thought.. what if we use allocation ID as the name & hostname of the container ?

Reply:
Yup. Makes sense.